config/tls.go

```diff
@@ -7,16 +7,51 @@ import (
 	"os"
 )
 
-// TLS provides TLS configuration options.
+// TLS represents configuration for a TLS client.
+// It provides options to enable TLS, specify certificate and key files,
+// CA certificate, and whether to skip verification of the server's certificate chain and host name.
+// Use the [TLS.MakeConfig] method to assemble a [*tls.Config] from the TLS struct.
+//
+// Example usage:
+//
+//	func main() {
+//		tlsConfig := &config.TLS{
+//			Enable:   true,
+//			Cert:     "path/to/cert.pem",
+//			Key:      "path/to/key.pem",
+//			Ca:       "path/to/ca.pem",
+//			Insecure: false,
+//		}
+//
+//		cfg, err := tlsConfig.MakeConfig("example.com")
+//		if err != nil {
+//			log.Fatalf("error creating TLS config: %v", err)
+//		}
+//
+//		// ...
+//	}
 type TLS struct {
-	Enable   bool   `yaml:"tls"`
-	Cert     string `yaml:"cert"`
-	Key      string `yaml:"key"`
-	Ca       string `yaml:"ca"`
-	Insecure bool   `yaml:"insecure"`
+	// Enable indicates whether TLS is enabled.
+	Enable bool `yaml:"tls"`
+
+	// Cert is the path to the TLS certificate file.
+	Cert string `yaml:"cert"`
+
+	// Key is the path to the TLS key file.
+	Key string `yaml:"key"`
```

Comment:
Please document the dependency linkage between these two fields, `.Cert != ""` $\Leftrightarrow$ `.Key != ""`.